scripts/build.sh

```diff
@@ -0,0 +1,55 @@
+#!/bin/bash
+
+set -eu 
+SCRIPT_DIR=$( cd -- "$( dirname -- "${BASH_SOURCE[0]}" )" &> /dev/null && pwd )
+source ${SCRIPT_DIR}/vars.sh
+
+BUILDDIR="$2"
+
+# build docker images and local binaries
+while getopts sghir flag; do
+    case "${flag}" in
+        s) echo "Building Stride Docker...  ";
+           docker build --tag stridezone:stride -f Dockerfile.stride . ;
+
+           printf '%s' "Building Stride Locally...  ";
```

Comment:
Why do we build both on Docker _and_ Locally? Because local gets copied over to docker?

Reply:
Docker just provides the runtime environment, but all commands are executed locally through local binaries. And then the state is bridged with docker volumes which is a way to share the file system between a local machine and the docker container.
So for instance, when running locally, we point `--home` to `scripts/state/stride1`, and when inside the docker container, we point `--home` to `~/.stride` - but those two folders are actually mirror images of each other